### PR TITLE
ci: parallelize linting and tests

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR splits linting, testing, and building into 3 separate Github Actions. Previously, building wasn't tested at all. This makes CI run faster, since it's now parallelized, saving up to about a minute of time.

### Context of Change

Faster CI

### Type of Change

- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

### TypeScript/Hooks Update

- [ ] Updated files to React Hooks
- [ ] Updated files to TypeScript

## Test Plan

CI passes.

### CODE REVIEW

File: src/containers/Transactions/index.jsx [hunk 1]
```diff
@@ -44,23 +44,24 @@ const Transaction = (props) => {
...
+  const short = identifier.substr(0, 8)
+  document.title = `${t('xrpl_explorer')} | ${t(
+    'transaction_short',
+  )} ${short}...`
 
   useEffect(() => {
-    const hash = data.raw ? data.raw.hash : undefined
-    const short = identifier.substr(0, 8)
-
-    document.title = `${t('xrpl_explorer')} | ${t(
```
Comment (ckniffen) [inline src/containers/Transactions/index.jsx:52]: We should still update title.

Comment (mvadari) [reply src/containers/Transactions/index.jsx:52]: It's updated outside of the `useEffect` now (line 49). Should it stay in the useEffect?

Comment (ckniffen) [reply src/containers/Transactions/index.jsx:52]: Sorry I missed that.  It is fine outside of useEffect.  Just check when switching to a different TX it still updates

Comment (mvadari) [reply src/containers/Transactions/index.jsx:52]: Confirmed that it does

**Review by pdp2121 — APPROVED** [review]

**Review by ckniffen — APPROVED** [review]

**Pull request merged**